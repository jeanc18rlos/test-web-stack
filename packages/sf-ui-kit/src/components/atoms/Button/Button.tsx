import { memo } from 'react';
import StyledButton from './styled/StyledButton';

export const colors = {
  cultured: '#f8f8f8',
  black: '#000000',
  white: '#FFFFFF'
};
export interface ButtonProps {
  variant: keyof typeof ButtonVariants;
}

interface Props {
  children: React.ReactNode;
}

export enum ButtonVariants {
  'primary',
  'secondary'
}

interface VariantProps {
  border: string;
  background: string;
  color: string;
}
export interface IVariants {
  primary: {
    default: VariantProps;
    hover: VariantProps;
    focus: VariantProps;
    disabled: VariantProps;
  };
  secondary: {
    default: VariantProps;
    hover: VariantProps;
    focus: VariantProps;
    disabled: VariantProps;
  };
}
export const variants: IVariants = {
  primary: {
    default: {
      border: '4px solid rgba(0, 0, 0, 0.1)',
      background: colors.white,
      color: colors.black
    },
    hover: {
      border: '4px solid rgba(0, 0, 0, 0.4)',
      background: colors.white,
      color: colors.black
    },
    focus: {
      border: '4px solid rgba(0, 0, 0, 0.5)',
      background: colors.white,
      color: colors.black
    },
    disabled: {
      border: '4px solid rgba(0, 0, 0, 0.1)',
      background: colors.white,
      color: 'rgba(0, 0, 0, 0.5)'
    }
  },
  secondary: {
    default: {
      border: '4px solid rgba(0, 0, 0, 0.1);',
      color: colors.black,
      background: 'transparent'
    },
    hover: {
      background: 'transparent',
      color: colors.black,
      border: '4px solid rgba(0, 0, 0, 0.4);'
    },
    focus: {
      background: 'transparent',
      border: '4px solid rgba(0, 0, 0, 0.5);',
      color: colors.black
    },
    disabled: {
      background: 'transparent',
      border: '4px solid rgba(0, 0, 0, 0.1);',
      color: 'rgba(0, 0, 0, 0.5)'
    }
  }
};

const Button = ({ children, ...rest }: Props & ButtonProps & React.HTMLAttributes<HTMLElement>) => (
  <StyledButton {...rest}>{children}</StyledButton>
);

export default memo(Button);
