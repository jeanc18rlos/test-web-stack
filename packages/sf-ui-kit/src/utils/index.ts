export { sizes, Typography } from './typography';
