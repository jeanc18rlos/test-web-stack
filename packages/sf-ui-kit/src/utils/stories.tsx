import styled from 'styled-components';
import { rem } from 'polished';

// Images
import logo from '../assets/superformula.png';
import { FC, HTMLProps, ReactNode } from 'react';

const StyledStoryHeading = styled.div`
  width: 100%;
  max-width: 85vw;
  margin: 2.5rem auto;
  text-transform: capitalize;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${({ theme }) => theme.fonts.primaryFont};
    color: ${({ theme }) => theme.colors.black};
    @media screen and (max-width: ${rem('768px')}) {
      font-size: ${rem('20px')};
    }
  }
  hr {
    border-color: ${({ theme }) => theme.colors.black};
  }
`;

export const StoryContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  width: 100%;
  > * {
    margin: 1rem;
  }
`;
export const StoryWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  align-items: center;

  th,
  td {
    font-weight: 400;
    font-family: ${({ theme }) => theme.fonts.primaryFont};
    padding: 0.5em 1em;
  }
`;

export const ComponentWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: 'row';
`;

const StyledStoryHeader = styled.div`
  margin: ${rem('8px')};
  width: calc(100% - ${rem('16px')});
  background: transparent;
  display: flex;
  justify-content: space-between;
  padding: 0 ${rem('50px')};
  border-radius: ${rem('8px')} ${rem('8px')} 0 0;
  position: relative;
  &::before {
    content: '';
    z-index: -1;
    border-top: 88px solid #18181a;
    border-left: 20px solid transparent;
    border-right: 20px solid transparent;
    height: 0;
    background: transparent;
    position: absolute;
    width: 100%;
    left: 0;
    top: 0;
    max-width: 100%;
    box-sizing: border-box;
    filter: drop-shadow(2px 2px 2px black);
  }
  img {
    position: absolute;
    right: 0;
    width: 125px;
    filter: drop-shadow(2px 4px 6px black);
  }
  box-sizing: border-box;
  h1 {
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
    font-family: ${({ theme }) => theme.fonts.primaryFont};
  }
  @media screen and (max-width: ${rem('768px')}) {
    &::before {
    content: '';
    z-index: -1;
    border-top: 50px solid #18181a;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;

  }
    h1 {
      font-size: ${rem('20px')};
    }
    img {
      width: ${rem('70px')};
    }
  }
`;

export const StoryHeading: FC<HTMLProps<ReactNode>> = ({ children }) => (
  <StyledStoryHeading>
    {children}
    <hr />
  </StyledStoryHeading>
);

export const StoryHeader: FC<HTMLProps<ReactNode> & { name: string }> = ({ name }) => (
  <StyledStoryHeader>
    <h1>{name}</h1>
    <img src={logo} />
  </StyledStoryHeader>
);
