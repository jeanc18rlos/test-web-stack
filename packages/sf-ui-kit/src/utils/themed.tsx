import { ReactNode } from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '../theme';

const themed = (component: ReactNode): renderer.ReactTestRenderer =>
  renderer.create(<ThemeProvider>{component}</ThemeProvider>);

export { themed };
