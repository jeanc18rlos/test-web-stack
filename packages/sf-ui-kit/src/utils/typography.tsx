import { css } from 'styled-components';
import { rem } from 'polished';

export const sizes = {
  h1: {
    size: `${rem('48px')}`,
    height: `${rem('60px')}`,
    weight: 300,
    style: 'normal'
  },
  h2: {
    size: `${rem('21px')}`,
    height: `${rem('26px')}`,
    weight: 600,
    style: 'normal'
  },
  body: {
    size: `${rem('16px')}`,
    height: `${rem('20px')}`,
    weight: 300,
    style: 'normal'
  }
};

enum Base {
  'h1',
  'h2',
  'body'
}

interface Options {
  size?: string;
  height?: number;
  weight?: number | string;
  style?: string;
  spacing?: string;
  color?: string;
  family?: string;
}

export const Typography = ({ base, options }: { base: keyof typeof Base; options?: Options }): any => css`
  font-feature-settings: 'cpsp' on;
  font-size: ${options && options.size ? options.size : sizes[base].size};
  line-height: ${options && options.height ? options.height : sizes[base].height};
  font-weight: ${options && options.weight ? options.weight : sizes[base].weight};
  font-style: ${options && options.style ? options.style : sizes[base].style};
  ${options && options.spacing && `letter-spacing: ${options && options.spacing};`}
  color: ${({ theme }) => (options && options.color ? options.color : theme.colors.black)};
  font-family: ${({ theme }) => (options && options.family ? options.family : theme.fonts.primaryFont)};
`;
