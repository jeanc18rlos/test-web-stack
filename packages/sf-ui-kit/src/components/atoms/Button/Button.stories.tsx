import { Meta } from '@storybook/react';
import Button from './Button';
import { StoryHeader, StoryHeading, StoryWrapper } from '../../../utils/stories';
import { variants } from './Button';
export default {
  title: 'Components/Atoms/Button',
  component: Button
} as Meta;

export const primary = () => {
  return (
    <StoryWrapper>
      <StoryHeader name={`Button`} />
      <StoryHeading>
        <h2>Primary</h2>
      </StoryHeading>
      <div
        style={{
          display: 'flex',
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'center'
        }}
      >
        {Object.keys(variants['primary']).map(key => {
          return (
            <div
              style={{
                padding: '8px'
              }}
              key={`button-${['primary']}-${[key]}`}
            >
              <Button variant={'primary'} className={`${key}`}>
                {key}
              </Button>
            </div>
          );
        })}
      </div>
    </StoryWrapper>
  );
};

export const secondary = () => {
  return (
    <StoryWrapper>
      <StoryHeader name={`Button`} />
      <StoryHeading>
        <h2>secondary</h2>
      </StoryHeading>
      <div
        style={{
          display: 'flex',
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'center'
        }}
      >
        {Object.keys(variants['secondary']).map(key => {
          return (
            <div
              style={{
                padding: '8px'
              }}
              key={`button-${['secondary']}-${[key]}`}
            >
              <Button variant={'secondary'} className={`${key}`}>
                {key}
              </Button>
            </div>
          );
        })}
      </div>
    </StoryWrapper>
  );
};
