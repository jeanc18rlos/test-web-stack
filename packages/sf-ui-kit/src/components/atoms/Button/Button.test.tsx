

import { themed } from '../../../utils/themed';
import Button from '.';

describe('Button', () => {
  it('should render correctly', () => {
    const component = themed(<Button variant={'primary'}>hello</Button>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
