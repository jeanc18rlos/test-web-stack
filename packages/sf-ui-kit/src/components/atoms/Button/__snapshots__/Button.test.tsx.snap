// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
.c0 {
  cursor: pointer;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background,border,color 0.25s ease-in-out;
  transition: background,border,color 0.25s ease-in-out;
  padding: 0.6875rem 0.875rem;
  max-width: 100%;
  max-height: 100%;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  min-height: -webkit-fit-content;
  min-height: -moz-fit-content;
  min-height: fit-content;
  width: 12.5rem;
  height: 4.375rem;
  font-feature-settings: 'cpsp' on;
  font-size: 1rem;
  line-height: 1.25rem;
  font-weight: 600;
  font-style: normal;
  color: #000000;
  font-family: Source Sans Pro,sans-serif;
  text-transform: uppercase;
  border: 4px solid rgba(0,0,0,0.1);
  color: #000000;
  background: #FFFFFF;
}

.c0.hover,
.c0:hover {
  border: 4px solid rgba(0,0,0,0.4);
  color: #000000;
  background: #FFFFFF;
}

.c0.focus,
.c0:focus {
  border: 4px solid rgba(0,0,0,0.5);
  color: #000000;
  background: #FFFFFF;
}

.c0.disabled {
  pointer-events: none;
  border: 4px solid rgba(0,0,0,0.1);
  color: rgba(0,0,0,0.5);
  background: #FFFFFF;
}

@media (max-width:48em) {
  .c0 {
    width: 9.375rem;
    height: 3.75rem;
  }
}

<button
  className="c0"
>
  hello
</button>
`;
